src/lib/sql/sql_translator.cpp

```diff
@@ -354,28 +402,45 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_join(const hsql::Join
   auto left_node = _translate_table_ref(*join.left);
   auto right_node = _translate_table_ref(*join.right);
 
-  const hsql::Expr& condition = *join.condition;
-
-  Assert(condition.type == hsql::kExprOperator, "Join condition must be operator.");
-  // The Join operators only support simple comparisons for now.
-  switch (condition.opType) {
-    case hsql::kOpEquals:
-    case hsql::kOpNotEquals:
-    case hsql::kOpLess:
-    case hsql::kOpLessEq:
-    case hsql::kOpGreater:
-    case hsql::kOpGreaterEq:
-      break;
-    default:
-      Fail("Join condition must be a simple comparison operator.");
+  auto get_side = [&left_node, &right_node](const hsql::Expr* expr) {
+    auto qualified = HSQLExprTranslator::to_qualified_column_name(*expr);
+    if (left_node->find_column(qualified)) {
+      return LQPChildSide::Left;
+    }
+
+    if (right_node->find_column(qualified)) {
+      return LQPChildSide::Right;
+    }
+
+    Fail("Column not found.");
+  };
+
+  std::vector<const hsql::Expr *> joins, left, right;
+  std::tie(joins, left, right) = _classify_join_conditions(join, get_side);
+
+  Assert(!joins.empty(), "No join condition supplied.");
+  Assert(joins.size() == 1, "More than one possible join condition supplied.");
```

Comment:
...which is only a problem for outer joins, you could include that in the message. But multiple Join predicates wasn't supported before as well, right?

Reply:
True.